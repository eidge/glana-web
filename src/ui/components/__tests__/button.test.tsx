import Button from "../button";
import { render, screen } from "@testing-library/react";

async function expectButtonToMatchSnapshot() {
  const button = await screen.findByRole("button");
  expect(button).toMatchSnapshot();
}

describe("Button", () => {
  it("renders solo icon", async () => {
    render(<Button size="lg" color="white" icon="cog" onClick={jest.fn} />);
    const icon = screen.queryByLabelText("cog");
    expect(icon).toBeVisible();
    expectButtonToMatchSnapshot();
  });

  it("renders solo text", async () => {
    render(
      <Button size="lg" color="white" text="click me" onClick={jest.fn} />
    );

    const text = await screen.findByText("click me");
    expect(text).toHaveTextContent("click me");
    expect(text).not.toHaveAttribute("class", expect.stringContaining("pl-2"));

    const icon = screen.queryByLabelText("cog");
    expect(icon).toBeNull();

    expectButtonToMatchSnapshot();
  });

  it("renders icon and text", async () => {
    render(
      <Button
        size="lg"
        color="white"
        icon="cog"
        text="click me"
        onClick={jest.fn}
      />
    );

    const text = await screen.findByText("click me");
    expect(text).toHaveTextContent("click me");
    expect(text).toHaveAttribute("class", expect.stringContaining("pl-2"));

    const icon = screen.queryByLabelText("cog");
    expect(icon).toBeVisible();

    expectButtonToMatchSnapshot();
  });
});
