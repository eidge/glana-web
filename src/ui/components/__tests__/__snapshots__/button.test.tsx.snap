// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders icon and text 1`] = `
<button
  class="jsx-943652769 gl-button inline-flex items-center justify-center leading-none p-2 focus:outline-none rounded shadow hover:shadow-inner text-white hover:bg-white hover:bg-opacity-10 "
  type="button"
>
  <i
    aria-label="cog"
    class="inline-block  w-8 h-8"
  >
    <svg
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10.325 4.317c.426-1.756 2.924-1.756 3.35 0a1.724 1.724 0 002.573 1.066c1.543-.94 3.31.826 2.37 2.37a1.724 1.724 0 001.065 2.572c1.756.426 1.756 2.924 0 3.35a1.724 1.724 0 00-1.066 2.573c.94 1.543-.826 3.31-2.37 2.37a1.724 1.724 0 00-2.572 1.065c-.426 1.756-2.924 1.756-3.35 0a1.724 1.724 0 00-2.573-1.066c-1.543.94-3.31-.826-2.37-2.37a1.724 1.724 0 00-1.065-2.572c-1.756-.426-1.756-2.924 0-3.35a1.724 1.724 0 001.066-2.573c-.94-1.543.826-3.31 2.37-2.37.996.608 2.296.07 2.572-1.065z"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
      <path
        d="M15 12a3 3 0 11-6 0 3 3 0 016 0z"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </i>
  <span
    class="jsx-943652769 leading-none text-2xl pl-2"
  >
    click me
  </span>
</button>
`;

exports[`Button renders solo text 1`] = `
<button
  class="jsx-943652769 gl-button inline-flex items-center justify-center leading-none p-2 focus:outline-none rounded shadow hover:shadow-inner text-white hover:bg-white hover:bg-opacity-10 "
  type="button"
>
  <span
    class="jsx-943652769 leading-none text-2xl"
  >
    click me
  </span>
</button>
`;
